### DISCUSSION TITLE
V2/routes endpoint with lat/lon/radius is timing out

### DISCUSSION THREAD

**Discussion by wiinci** [transitland/transitland#342] [Q&A] [answered]
Hello, I'm seeing 504 timeout errors while using the[ new V2 routes endpoint](https://www.transit.land/documentation/rest-api/routes). Please advise in case I'm missing something. Thanks! 

query:
https://api.transit.land/api/v2/rest/routes?apikey={{key}}&radius=400&lat=33.73785665117274&lon=-84.30342366129413

response:
```json
{
  "message": "The upstream server is timing out"
}
```

<details>
<summary>
Details:
</summary>

```
Request
:method
:scheme: https
:authority: api.transit.land
:path: /api/v2/rest/routes?apikey={{key}}&radius=300&lat=33.73785665117274&lon=-84.30342366129413
```

```
Response
:status: 504
Content-Type: application/json; charset=utf-8
Access-Control-Allow-Origin: *
Via: kong/2.4.1
Date: Thu, 04 Nov 2021 01:04:47 GMT
Content-Length: 51
Vary: Origin
ratelimit-remaining: 59
x-ratelimit-limit-minute: 60
x-kong-upstream-latency: 60002
Server: openresty
ratelimit-limit: 60
x-ratelimit-remaining-month: 9895
x-ratelimit-limit-month: 10000
x-ratelimit-remaining-minute: 59
ratelimit-reset: 13
x-kong-proxy-latency: 300005
```

```
Query String Parameters
apikey: {{key}}
radius: 400
lat: 33.73785665117274
lon: -84.30342366129413
```
</details>

**Comment by irees** [comment]
@wiinci

The route lat/lon search is a little complicated behind the scenes, since it queries based on stop locations that are associated with a route. I'll take a look and see if there is a reason this particular location fails and if it can be optimized further.

**Reply by boredland** [reply]
this problem seems to persist.. did you have a look already? Here is another example: 
https://transit.land/api/v2/rest/routes?lat=40.725&lon=-73.996&radius=10&include_geometry=true&limit=10&api_key={}

**Reply by irees** [reply]
Taking a look now... looks like a particular `where` clause is on the outside of a subquery when it needs to be on the inside to be performant. Oops.

**Reply by irees** [reply]
I have an improved approach to route geometry searches; testing locally and then will deploy. I'll give you an update when it's done.

**Comment by irees** [comment, marked as answer]
@boredland This query should now complete in a reasonable amount of time. (Note: a larger radius used)

https://transit.land/api/v2/rest/routes?lat=40.725&lon=-73.996&radius=1000&include_geometry=true&limit=10&api_key={}

**Reply by boredland** [reply]
The OP query is reasonably fast now, thanks! The one you posted took a huge amount of time (sadly didn't have my network tab open, but easily 30s) initially, but re-fetches are fast (which may be caching only). So not 100% sure you fixed it completely? What speed would you expect there?
